src/InvestJob.jsx

```diff
@@ -37,9 +35,36 @@ const LOGLEVELMAP = {
   ERROR: '-v',
 };
 
-function changeSetupKey(int) {
-  // Used for generating a new unique `key` for SetupTab component.
-  return (int + 1);
+/** Get an invest model's ARGS_SPEC when a model button is clicked.
+ *
+ * Also get the model's UI spec if it exists.
+ *
+ * @param {string} modelName - as in a model name appearing in `invest list`
+ * @returns {object} destructures to:
+ *   { modelSpec, argsSpec, uiSpec }
+ */
+async function investGetSpec(modelName) {
+  const spec = await getSpec(modelName);
+  if (spec) {
+    const { args, ...modelSpec } = spec;

```

Comment:
The `...` syntax is cool! So `args` is `spec['args']` is the `ARGS_SPEC`, and `modelSpec` has any other keys with info about the model?

Reply:
Yes, this is spread syntax! Recommended by our linter. 

`{ args }` is `spec['args']` is `ARGS_SPEC['args']` and `modelSpec` is the rest!